k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java

```diff
@@ -1489,12 +1533,28 @@ public void onMessageBuildSuccess(MimeMessage message, boolean isDraft) {
             }
         } else {
             currentMessageBuilder = null;
+
+            if (mAction == Action.REDIRECT) {
+                LocalMessage referenceMessage = mMessageReference.restoreToLocalMessage(getApplicationContext());
+                message = referenceMessage.cloneAsSuper(true);
+                message.setResentDate(new Date(), K9.hideTimeZone());
+                message.setRecipients(RecipientType.RESENT_TO, toArray(recipientPresenter.getToAddresses()));
```

Comment:
i haven't found any hint about the RCPT being wrong - nor about it would matter at all when using resent-headers.
do you have a server where you could test the behaviour?